src-test/com/zone5cloud/retrofit/core/TestOAuthAPI.java

```diff
@@ -243,4 +244,58 @@ public void testAdhocToken() {
 			assertNotNull("Token should havea  scope", token.getScope());
 		}
 	}
+
+	@Test
+	public void testZone5Server_HasClientIdAndKey(){
+		Response<User> rsp = userApi.me().blockingFirst();
+		String clientId = rsp.raw().request().header("Api-Key");
+		String clientSecret = rsp.raw().request().header("Api-Key-Secret");
+
+		System.out.println("Client Id :"+clientId + "Client Secret : "+clientSecret);
+		assertNotNull(clientId);
+		assertNotNull(clientSecret);
+	}
+
+	@Test
+	public void testNonZone5Server_HasNoClientIdAndKey(){
+		Response<User> rsp = userApi.me().blockingFirst();
+		String clientId = rsp.raw().request().header("Api-Key");
+		String clientSecret = rsp.raw().request().header("Api-Key-Secret");
+
+		System.out.println("Client Id :"+clientId + "Client Secret : "+clientSecret);
+		assertNull(clientId);
+		assertNull(clientSecret);
+	}
```

Comment:
what is different between testZone5Server_HasClientIdAndKey() and testNonZone5Server_HasNoClientIdAndKey()? They are exactly the same but expect different outcomes.

the BaseTest retrofit instance is configured with base url of getBaseEndpoint() which uses the enviornment variable server in the env file. This is the same url that needs to be passed to the authorization interceptor as the Zone5BaseURL. I don't think you needs a separate environment variable.

To test the non-zone5 case we need a second retrofit instance (with the same auth interceptor), which can just be instantiated in the specific test case, that is configured to be a different host url - and not necessarily a zone5 one. For example you could configure it with "https://developer.apple.com" and then create an api in the test case:

```
public interface TestExternalAPI {
@GET("/augmented-reality/quick-look/models/gramophone/gramophone.usdz")
Observable<URL> downloadExternal();
}

TestExternalAPI testAPI = retrofitExternal.create(TestExternalAPI.class);
```
 
calling testAPI.downloadExternal().blocklingSingle() will attempt to send a request to apple and you can assert that the tp headers are not added to that.

To test autorefresh, manually set the token expiry to be expired. Run the external api request. Check that the auth token has been updated. In spite of the request going to an external server, the refresh request needs to go to the configured zone5BaseURL.

Please ask if that doesn't make sense

Reply:
> what is different between testZone5Server_HasClientIdAndKey() and testNonZone5Server_HasNoClientIdAndKey()? They are exactly the same but expect different outcomes.

> The first case is checking when it is a zone5 server the api key and secret are attached and the other is to test they are not when it is not zone5 (changing the server url param in env variable)

> the BaseTest retrofit instance is configured with base url of getBaseEndpoint() which uses the enviornment variable server in the env file. This is the same url that needs to be passed to the authorization interceptor as the Zone5BaseURL. I don't think you needs a separate environment variable.

> ok will change it to use the method to retrieve base endpoint

> To test the non-zone5 case we need a second retrofit instance (with the same auth interceptor), which can just be instantiated in the specific test case, that is configured to be a different host url - and not necessarily a zone5 one. For example you could configure it with "https://developer.apple.com" and then create an api in the test case:
> ok
> ```
> public interface TestExternalAPI {
> @GET("/augmented-reality/quick-look/models/gramophone/gramophone.usdz")
> Observable<URL> downloadExternal();
> }
> 
> TestExternalAPI testAPI = retrofitExternal.create(TestExternalAPI.class);
> ```
> 
> calling testAPI.downloadExternal().blocklingSingle() will attempt to send a request to apple and you can assert that the tp headers are not added to that.
> 
> To test autorefresh, manually set the token expiry to be expired. Run the external api request. Check that the auth token has been updated. In spite of the request going to an external server, the refresh request needs to go to the configured zone5BaseURL.
> 
> Please ask if that doesn't make sense



> what is different between testZone5Server_HasClientIdAndKey() and testNonZone5Server_HasNoClientIdAndKey()? They are exactly the same but expect different outcomes.
> 
> the BaseTest retrofit instance is configured with base url of getBaseEndpoint() which uses the enviornment variable server in the env file. This is the same url that needs to be passed to the authorization interceptor as the Zone5BaseURL. I don't think you needs a separate environment variable.
> 
> To test the non-zone5 case we need a second retrofit instance (with the same auth interceptor), which can just be instantiated in the specific test case, that is configured to be a different host url - and not necessarily a zone5 one. For example you could configure it with "https://developer.apple.com" and then create an api in the test case:
> 
> ```
> public interface TestExternalAPI {
> @GET("/augmented-reality/quick-look/models/gramophone/gramophone.usdz")
> Observable<URL> downloadExternal();
> }
> 
> TestExternalAPI testAPI = retrofitExternal.create(TestExternalAPI.class);
> ```
> 
> calling testAPI.downloadExternal().blocklingSingle() will attempt to send a request to apple and you can assert that the tp headers are not added to that.
> 
> To test autorefresh, manually set the token expiry to be expired. Run the external api request. Check that the auth token has been updated. In spite of the request going to an external server, the refresh request needs to go to the configured zone5BaseURL.
> 
> Please ask if that doesn't make sense